payment-account-age-witness.adoc

```diff
@@ -0,0 +1,196 @@
+
+We suggest following limits:
+
+. Account never used in trade: 25% of the default limit (e.g. 0.125 BTC for a default limit of 0.5 BTC)
+. Trade age < 1 months: 50% of the default limit (e.g. 0.25 BTC for a default limit of 0.5 BTC)
+. Trade age 1 - 2 months: 75% of the default limit (e.g. 0.375 BTC for a default limit of 0.5 BTC)
+. Trade age > 2 months: Default limit will be applied (e.g. 0.5 BTC)
+
+_Please note that in future the trade amount limits will be probably derived from the current market price and the target will be likely about 1000 USD._
+
+
+== AgeWitness
+
+When a user makes his *first trade* with a certain Fiat payment account (e.g. SEPA, Zelle,...) he publishes a new object (we call it AgeWitness) to the P2P network.
+
+_Side note:
+The reason why we separate that into a new data structure and don't use the already existing trade statistics data structure is because in future, trade statistic data might get pruned once the historical data gets too large. We don't want to be exposed to the complexity introduced in future pruning solutions nor do we want to add constraints to that pruning. We can assume that the AgeWitness data will never become that large that pruning will become necessary._
+
+
+=== AgeWitness data object
+
+The AgeWitness object contains a hash, a signature, the pubKey and the tradeDate. The hash gets created with Sha256. Input is the ageWitnessInputData concatenated with a 256 bytes salt. The ageWitnessInputData is the smallest set of uniquely identifying payment account data (e.g. concatenated IBAN and BIC). We don't use the complete payment account data because we don't want to break an existing ageWitness by minor changes like changing the holder name (e.g. adding middle name).
+
+We add also a signature of the hash and the public key for assuring that the AgeWitness data cannot be used by anyone else (see: <<hijacking>>). The application wide 1024 bit DSA signing key is used. Signature algorith is "SHA256withDSA".
+
+The tradeDate must not be older than 1 day compared to receiving peer's local date. If the data is too old the AgeWitness object will get rejected. With that we protect against back-dating attempts (see: <<back-dating>>). We allow a rather large tolerance because computer clocks might be out of sync and the relevant periods are rather long (1 month, 2 months), so the max. gain from an abuse of that tolerance window of 1 day is negligible.
+
+----
+    // class PaymentAccountAgeWitnessService
+    public PaymentAccountAgeWitness getPaymentAccountWitness(PaymentAccount paymentAccount, Trade trade) throws CryptoException {
+        byte[] ageWitnessInputData = paymentAccount.getAgeWitnessInputData();
+        byte[] salt = paymentAccount.getSalt();
```

Comment:
getSalt(32) or don't define the getSalt with a parameter

Reply:
Its created in the init method of the PaymentAccount with that param.